docs/berkeley-upgrade/planning-archive-migration.mdx

```diff
@@ -80,84 +132,89 @@ If you don't have an existing database with Mainnet archive data, you can always
 
-```sh
-mina-replayer --archive-uri {db_connection_string} --input-file reference_replayer_input.json --output-file reference_replayer_output.json --checkpoint-interval 10000
+To get the Docker image:
+
+
+```
+docker pull gcr.io/o1labs-192920/mina-archive-berkeley-archive-migration:1.0.0umt-tooling
 ```
 
-where:
- - `reference_replayer_input.json` is a file constructed our of mainnet genesis ledger:
-  
-   ```
-   echo '{ "genesis_ledger": { "accounts": [ ], "add_genesis_winner" : true } }' > input-file.json
+## Devnet/Mainnet Genesis Ledger
 
-   jq --argfile genesis ./genesis_ledger_mainnet.json '.genesis_ledger.accounts += $genesis.ledger.accounts' ./input-file.json > replayer-reference-input.json
-   ```
+The Mina Devnet/Mainnet genesis ledger is stored in GitHub in the `mina` repository under the `genesis_ledgers` subfolder. However, if you are already running a daemon that is connected to the Mina Mainnet or the Devnet network, you already have the genesis ledger locally.
 
-:warning: Running a replayer from scratch on a Mainnet database can take up to a couple of days. The recommended best practice is to break the replayer into smaller parts by using the checkpoint capabilities of the replayer.
+## Berkeley database schema files
 
-:warning: You must run replayer from the Mainnet version. You can run it from the Docker image at minaprotocol/mina-archive:1.4.0-c980ba8-bullseye
+You can get the Berkeley schema files from different locations:
 
-### Google Cloud bucket with Mainnet precomputed blocks
+- GitHub repository under `berkeley` branch. 
 
-The Mainnet to Berkeley archive data migration requires access to precomputed blocks that are uploaded by daemons connected to the Mainnet network. The Berkeley migration app uses the gsutil app to download blocks. If you didn't store precomputed blocks during first phase of migration, you can use the precomputed blocks provided by Mina Foundation:
+    Note: The `berkeley` branch can contain new updates regarding schema files, so always get the latest schema files instead of using an already downloaded schema. 
+- Archive/Rosetta Docker from `berkeley` version
 
-`gsutil cp gs://mina_network_block_data/mainnet-*.json .`
+### Example: Downloading schema sources from GitHub
 
-:warning: Precomputed blocks for 'mainnet' network takes ~800Gb. Downloading them make take some time 
+   ```sh
+   wget https://raw.githubusercontent.com/MinaProtocol/mina/berkeley/src/app/archive/zkapp_tables.sql
 
+   wget https://raw.githubusercontent.com/MinaProtocol/mina/berkeley/src/app/archive/create_schema.sql
 
-The best practice is to collect precomputed blocks by yourself or by other third parties to preserve idea of decentralization. 
+   ```
 
-## Migration process
+## Next steps
 
 You are now ready to perform migration. See:
 
-- [How to perform archive migration](/berkeley-upgrade/migrating-archive-database-to-berkeley)
+- [How to perform archive migration](/berkeley-upgrade/planning-archive-migration)
```

Comment:
```suggestion
- [How to perform archive migration](/berkeley-upgrade/planning-archive-migration)

## Worked Example for March 22 data

```sh
	sudo rm /etc/apt/sources.list.d/mina*.list
	sudo echo "deb [trusted=yes] http://packages.o1test.net $(lsb_release -cs) unstable" | sudo tee /etc/apt/sources.list.d/mina.list
	sudo apt-get update

	sudo apt-get install --allow-downgrades -y mina-archive-migration=3.0.0-rc0
	
	mkdir mina-migration-workdir
	cd mina-migration-workdir
	
	gsutil cp gs://mina_network_block_data/devnet-bundle-3NKRsRWBzmPR8Z8ZmJb4u8FLpnSkjRitUpKZzVkHp11QuwP5i839.tar.gz .
	tar -xf devnet-bundle-3NKRsRWBzmPR8Z8ZmJb4u8FLpnSkjRitUpKZzVkHp11QuwP5i839.tar.gz
	
	wget https://raw.githubusercontent.com/MinaProtocol/mina/berkeley/src/app/archive/create_schema.sql
	wget https://raw.githubusercontent.com/MinaProtocol/mina/berkeley/src/app/archive/zkapp_tables.sql
	
	# this next step is only needed if you don't have an archive yet
	createdb devnet_balances_migrated
	createdb devnet_really_migrated
	
	psql -d devnet_really_migrated -f create_schema.sql
	
	gsutil cp gs://mina-archive-dumps/devnet-archive-dump-2024-03-22_0000.sql.tar.gz .
	tar -xf devnet-archive-dump-2024-03-22_0000.sql.tar.gz
	# the next step ensures you don't accidentally merge mainnet and devnet data
	sed -i -e s/archive_balances_migrated/devnet_balances_migrated/g devnet-archive-dump-2024-03-22_0000.sql
	psql -d devnet_balances_migrated -f devnet-archive-dump-2024-03-22_0000.sql
	
	mina-berkeley-migration-script initial \
	    --genesis-ledger /var/lib/coda/devnet.json \
	    --source-db postgres:///devnet_balances_migrated \
	    --target-db postgres:///devnet_really_migrated \
	    --blocks-batch-size 100 --blocks-bucket mina_network_block_data \
	    --network devnet


	# now, do a final migration
	
	gsutil cp gs://mina-archive-dumps/devnet-archive-dump-2024-03-22_2050.sql.tar.gz .
	tar -xf devnet-archive-dump-2024-03-22_2050.sql.tar.gz
	# the next step ensures you don't accidentally merge mainnet and devnet data
	sed -i -e s/archive_balances_migrated/devnet_balances_migrated/g devnet-archive-dump-2024-03-22_2050.sql
	psql -d devnet_balances_migrated -f devnet-archive-dump-2024-03-22_2050.sql
	
	curl -O  http://fs.hbox.lunar.town/fork-devnet-march-22-2024.json
	
	mina-berkeley-migration-script final \
	    --genesis-ledger /var/lib/coda/devnet.json \
	    --source-db postgres:///devnet_balances_migrated \
	    --target-db postgres:///devnet_really_migrated \
	    --blocks-batch-size 100 --blocks-bucket mina_network_block_data \
	    --network devnet \
	    --replayer-checkpoint migration-checkpoint-437195.json \
	    --fork-state-hash 3NKoUJX87VrfmNAoUdqoWUykVvt66ztm5rzruDQR7ihwYaWsdJKq \
	    --fork-config fork-devnet-march-22-2024.json \
	    --prefetch-blocks

 ```